libs/blocks/quiz/utils.js

```diff
@@ -152,41 +156,73 @@ const structuredFragments = (
   return structureFragments;
 };
 
-const nestedFragments = (
-  nestedFragsArray,
+export const nestedFragments = (
+  nestedFragsPrimaryArray,
+  nestedFragsSecondaryArray,
   resultResources,
   primaryProducts,
-  secondaryProductCodes,
+  secondaryProducts,
   isUmbrella,
 ) => {
   const nestedObject = {};
-  nestedFragsArray.forEach((frag) => {
+  nestedFragsPrimaryArray?.forEach((frag) => {
+    if (!frag) return;
     const fragKey = frag.trim();
-    const fragArray = [];
-    resultResources.data.forEach((row) => {
-      if (isUmbrella) {
-        // Get nested frags for all the primary products.
-        if (primaryProducts.length > 0 && primaryProducts.includes(row.product)) {
-          if (row[fragKey]) {
-            fragArray.push(row[fragKey]);
-          }
-        }
-      } else if (secondaryProductCodes.length > 0 && secondaryProductCodes.includes(row.product)) {
-        // Get nested frags for all the secondary products.
-        if (row[fragKey]) {
-          fragArray.push(row[fragKey]);
-        }
-      }
-    });
-    nestedObject[fragKey] = fragArray;
+    // eslint-disable-next-line max-len
+    nestedObject[fragKey] = getNestedFragments(resultResources, isUmbrella, primaryProducts, secondaryProducts, fragKey, true);
```

Comment:
Since this has 6 arguments, it might be better to use an object. Changing anything becomes difficult, since there's a strict order and it's difficult to understand what certain arguments refer to. What does `true` or `false` achieve? Something like
```
getNestedFragments({
  resultResources,
  isUmbrella,
  primaryProducts,
  secondaryProducts,
  fragKey,
  fetchForPrimaryProducts: true,
});
```
is a bit more clear and more maintainable, as you could omit certain arguments, change their order, etc.

Reply:
Refactored the entire way we are fetching nested frags which has surely simplified the flow a lot. Would be great if you want to have a second look.